Cargo.toml

```diff
@@ -33,9 +34,15 @@ rstest = "0.17.0"
 test_fiat_shamir = []
 instruments = []                   # This enables timing prints in prover and verifier
 metal = ["lambdaworks-math/metal"]
+rayon = ["dep:rayon"]
```

Comment:
I would suggest to re-name the feature to something like `parallel` instead of `rayon`, which is the name of the crate. It will be simpler for users to understand what this feature does + maybe in the future another dependency is used rather than rayon. What do you think?